webClient/src/app/shared/editor-control/editor-control.service.ts

```diff
@@ -583,6 +584,126 @@ export class EditorControlService implements ZLUX.IEditor, ZLUX.IEditorMultiBuff
     });
   }
 
+  public isContentValidForDataset(content: string[], datasetAttrs: DatasetAttributes): boolean | string {
+    //TODO: validation of record length that is aware of how DBCS will effect actual length
+    //FB must have exactly lrecl, VB must have no more than lrecl. content cant be undefined.
+    const MAX_CONTENT_LENGTH = 5242880;
+    if (!content) {
+      return 'No Content';
+    }
+    let maxRecordLen = datasetAttrs.dsorg.maxRecordLen;
+    for (let i = 0; i < content.length; i++) {
+      let record = content[i];
+      if (record.length > maxRecordLen) {
+        return `Line ${i} exceeds record length limit of ${maxRecordLen}`;
```

Comment:
`i `is 1 off the value you need

![image](https://user-images.githubusercontent.com/20528015/134260985-e3725152-a27c-4e87-8138-3f43afd22c8a.png)